core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -283,7 +283,7 @@ class BrokerToControllerRequestThread(
   time: Time,
   threadName: String,
   retryTimeoutMs: Long
-) extends InterBrokerSendThread(threadName, networkClient, config.controllerSocketTimeoutMs, time, isInterruptible = false) {
+) extends InterBrokerSendThread(threadName, networkClient, Math.min(Int.MaxValue, Math.min(config.controllerSocketTimeoutMs, retryTimeoutMs)).toInt, time, isInterruptible = false) {
```

Comment:
let's just use `retryTimeoutMs`, there is no need to involve `controllerSocketTimeoutMs`. That is not a KRaft config anyway...